### ISSUE TITLE
Cannot find module 'encoding/package.json'

### ISSUE THREAD

**Issue by ryanjwilke** [netlify/netlify-lambda#201]
I noticed that someone else a few months ago was struggling with this issue as well, however it wasn't clear what actually resolved the issue for this person. Do you have any idea why this works and builds fine locally, but the build fails in production with the following error? I'm using Netlify Dev as well so theoretically the environments should be pretty close.

```
5:58:42 PM: Error: In file "/opt/build/repo/lambda-server/user-create.js": Cannot find module 'encoding/package.json' from '/opt/build/repo/lambda-server'
```

Here's some evidence of the build failing at [vue-cli-template#27](https://github.com/ryanjwilke/vue-cli-template/pull/27)

**Comment by ryan-gilb** [comment]
Hey there 👋 

I was running into this issue as well. After some digging around, I found a workaround that worked for me! The issue for me is in `node-fetch`. Hopefully this can help you 😄

**TLDR**

> You should:
> * npm install encoding
> * add require('encoding'); statements in the function files that use node-fetch

Peep this comment, it helped me tremendously: https://github.com/netlify/zip-it-and-ship-it/issues/67#issuecomment-549837499

---

A module my project depends on _conditionally imports_ the `encoding` module, and also **does not** include `encoding` in it's own `package.json`. This resulted in my build (with `netlify-lambda build`) to succeed, but the zip (with `@netlify/zip-it-and-ship-it`) to **not include** `encoding`, so the import would fail.

One annoying thing is that I had to add `encoding` to my top-level package, which is my frontend app. Adding `encoding` only to the `package.json` in my `functions` folder still fails, probably because that's not where `zip-it-and-ship-it` is looking? Not sure 😕 I'll probably end up giving `node-fetch` the boot and use a different request client that hopefully doesn't necessitate this workaround!

**Comment by ryanjwilke** [comment]
@ryan-gilb Thank you for the detailed answer. I'll give that a try soon! That sounds like a simple enough way to fix the issue.

**Comment by waynevanson** [comment]
HI Guys,

I just ran into this issue.

I fixed it by adding `encoding` to my `package.json`.  should `encoding` be part of the this package or of `@netlify/plugin-functions-core`?

Here is the error in my console.
```sh
[28] ./test-function.ts 707 bytes {1} [built]
    + 14 hidden modules

 ✔  build.lifecycle.onBuild completed in 10340ms

┌────────────────────────────────────────────────────────────────────┐
│ 3. Running onPostBuild command from @netlify/plugin-functions-core │
└────────────────────────────────────────────────────────────────────┘

Packaging functions from __temp/lambda

┌────────────────────────────────────────────────────────┐
│ Plugin "@netlify/plugin-functions-core" internal error │
└────────────────────────────────────────────────────────┘

❯ Error message
  Error: In file "/home/wayne/Documents/netlify-test/__temp/lambda/fauna-test.js": Cannot find module 'encoding/package.json' from '/home/wayne/Documents/netlify-test/__temp/lambda'​

❯ Plugin details
  ID:             @netlify/plugin-functions-core
  Version:        0.1.100
  npm link:       https://www.npmjs.com/package/@netlify/build​

❯ Error location
  In "onPostBuild" event in npm package "@netlify/plugin-functions-core"
      at /home/wayne/Documents/netlify-test/node_modules/resolve/lib/async.js:107:35
      at processDirs (/home/wayne/Documents/netlify-test/node_modules/resolve/lib/async.js:260:39)
      at isdir (/home/wayne/Documents/netlify-test/node_modules/resolve/lib/async.js:267:32)
      at /home/wayne/Documents/netlify-test/node_modules/resolve/lib/async.js:23:69
      at FSReqCallback.oncomplete (fs.js:158:21)​

❯ Error properties
  { code: 'MODULE_NOT_FOUND' }

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
